ecosystem/sep-0039.md

```diff
@@ -0,0 +1,182 @@
+
+Note also that this SEP does not address concerns around *immutability*. While there is an optional checksum component (addressed later), the account itself is free to modify the data at-will if it is not locked.
+
+Our exclusive focus in this SEP is Step 2: **storing the asset within the account.**
+
+### Data Format
+We begin by describing the raw data format, then dive into each component.
+
+    00|version|metadata length|media-type;param1=value;param2=value;etc<binary data>
+
+Here, `|` is not a character, but rather a semantic divider used to separate the components. The following sections break down each component in detail.
+
+#### Index
+Since an account can only have 1000 entries, we only need 10 bits for an index field. We need indices to be ASCII-sortable in Horizon, so the index is expressed as a "base 36" value: 0-9 then a-z.
+
+A valid SEP-39 entry always begins with `00`, indicating that this is the 0th row of data. In fact, every row's key _must_ begin with a two-character, [Base36] index.
+
+---------
+
+**Author's note**: This portion is definitely up for debate. Base36 (well, more accurately, a radix of 36) allows us to shave a byte off of each row (saving up to 1kB total), but we could stick to a 000-999 model (stringified integer) for simplicity. Note that supporting it is trivial in any language, however, since we know it's strictly an integer. For example,
+
+```python
+import string 
+ALPHABET = string.digits + string.ascii_lowercase
+
+def int_to_b36(i: int) -> str:
+    d, r = divmod(i, len(ALPHABET))
+    return ALPHABET[d] + ALPHABET[r]
+
+def b36_to_int(s: str) -> int:
+    return ALPHABET.index(s[0]) * len(ALPHABET) + ALPHABET.index(s[1])
+```
+
+Another consideration is perhaps trying to have **two** indices within this value. Since we need 10 bits, we can use the other 6 for identifying this asset within the account itself. There's still the question of printables, but  though, so this needs more thought.
+
+---------
+
+#### Version
+The first entry's index is followed by the version string. For the current version of this SEP, this is just `1`.
+
+#### Metadata
+The **metadata** is a [Media Type](https://www.iana.org/assignments/media-types/media-types.xhtml) alongside any associated parameters. It is prefixed by a **length**, which is just a stringified integer (that is, a 32-byte metadata will have "32" as its length). When parsing, you should consume characters until hitting a non-digit, since a valid Media Type will never start with a digit. Note that row indices are not included in the metadata length.
+
+**Author's note**: If we go with a radix of 36, we could use that here, too.
+
+There are no *required* parameters.
+
+Standard *optional* parameters include:
+
+  * `n`: this defines the name or "title" of your asset
+  * `d`: this provides a description your asset
+  * `c`: a crc32 checksum of the asset
```

Comment:
I've never heard of `crc32`. Could we get a little context here like you have for basE91?

Reply:
Sure, yeah. I'll start here: crc32 is a very common (definitely not esoteric like BasE91) checksum algorithm (see [Wikipedia](https://en.wikipedia.org/wiki/Cyclic_redundancy_check)) It's not designed with cryptographic principles (integrity, collision resistance, inversion, etc.) in mind like the SHA family is, but rather just with the goal of error-checking. It's a fast, simple way to check that you have the right data.